jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -167,6 +170,21 @@ def search(self, error_on_fail=False):
                             'Object ID': row['objid']}
                 self.table.add_item(row_info)
 
+        elif self.catalog_selected == 'Gaia':
+            from astroquery.gaia import Gaia
+
+            Gaia.ROW_LIMIT = self.max_rows
+            sources = Gaia.query_object(skycoord_center, radius=zoom_radius,
+                                        columns=('source_id', 'ra', 'dec'))
```

Comment:
Users might have the Gaia module open in their Python session before or after they run this catalog search. If you set the `ROW_LIMIT` attr here, it will persist elsewhere in the Python session, and could mess up the user. We can temporarily set the row limit in a context manager like this:
```suggestion
            from astroquery.gaia import conf
            
            with conf.set_temp("ROW_LIMIT", self.max_sources):
                sources = Gaia.query_object(skycoord_center, radius=zoom_radius,
                                            columns=('source_id', 'ra', 'dec'))
```
which @pllim taught me for MAST queries recently:
https://github.com/spacetelescope/jdaviz/blob/4fe79a9b90b846d50a967151ae129483f729aaa3/jdaviz/utils.py#L580-L583